### PR TITLE
Add distribution classes for AutoML

### LINKED ISSUE #1290: Add distribution classes for AutoML
### 🚀 Feature Request

Now we are using distribution classes from optuna, but all we need from them is their name and parameters. Using external classes complicates their integration into transforms and models.

### Proposal

- Add abstract `BaseDistribution` into `etna.auto.distributions`
- Add `CategoricalDistribution` with `choices` parameter
  - We should validate the types of `choices` like `optuna` does
- Add `IntDistribution` with parameters `low`, `high`, `log`
  - We should validate input parameters like `optuna` does
- Add `FloatDistribution` with parameters `low`, `high`, `log`
  - We should validate input parameters like `optuna` does
- Rewrite `params_to_tune` method to return `Dict[str, BaseDistribution]`
- Fix `Tune` to use new distributions
- Add section about distributions into AutoML notebook

### Test cases

- Fix current tests that work with `optuna` distributions
- Add tests on creation of new tests

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #1290.

## Closing issues
Closes #1290.

### CODE REVIEW

**Comment by Mr-Geekman** [comment]
We should discuss how we are going to make documentation for `etna.distributions` is we are going to keep it that way.

File: etna/auto/utils.py [hunk 1]
```diff
@@ -4,6 +4,20 @@
...
+from typing import Dict
+
+import optuna
+
+from etna.distributions import BaseDistribution
+from etna.distributions import CategoricalDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+
+_DISTRIBUTIONS = {
```
Comment (martins0n) [inline etna/auto/utils.py:16]: Do these lambdas work in distributed env?

Comment (Mr-Geekman) [reply etna/auto/utils.py:16]: I'll try to make smth more robust, because it is now dataclasses and it can be made much simpler.

Comment (Mr-Geekman) [reply etna/auto/utils.py]: Fixed.

File: etna/auto/utils.py [hunk 2]
```diff
@@ -24,3 +33,21 @@ def retry(func: Callable[..., Any], max_retries: int = 5, sleep_time: int = 10,
                 continue
             else:
                 raise e
+
+
+def suggest_parameters(trial: optuna.Trial, params_to_tune: Dict[str, BaseDistribution]) -> Dict[str, Any]:
```
Comment (martins0n) [inline etna/auto/utils.py:38]: Is it ok that we miss a half of existing distributions?

Comment (martins0n) [reply etna/auto/utils.py:38]: It seems ok

**Review by martins0n — APPROVED** [review]

**Pull request merged**